lib/matplotlib/backend_bases.py

```diff
@@ -3518,7 +3560,11 @@ def new_figure_manager_given_figure(cls, num, figure):
 
     @classmethod
     def draw_if_interactive(cls):
-        if cls.mainloop is not None and is_interactive():
+        manager_class = cls.FigureCanvas.manager_class
+        backend_is_interactive = (
+            manager_class.start_main_loop != FigureManagerBase.start_main_loop
```

Comment:
```suggestion
            # interactive backends reimplement either start_main_loop or pyplot_show
            manager_class.start_main_loop != FigureManagerBase.start_main_loop
```
Or should we put this into a class method so that we can do
```
manager_class.backend_is_interactive()
```
?

Reply:
I'd rather not.  First, this is basically the definition of interactiveness that matches the old behavior of draw_if_interactive (overriding start_main_loop or pyplot_show is essentially like overriding mainloop previously), but there's also a different definition, namely that manager.show doesn't raise a NonGuiException.  In practice, classes that override manager.show to not raise a NonGuiException are also those that override s_m_l/p_s, but I don't want to formalize this definition, in particular, because overriding s_m_l/p_s also only matters for draw_if_interactive, which is becoming "semi-deprecated" since #23291.